hbase-server/src/main/java/org/apache/hadoop/hbase/procedure2/store/region/RegionProcedureStore.java

```diff
@@ -308,63 +308,69 @@ private void tryMigrate(FileSystem fs) throws IOException {
-          corruptedCount++;
+        @Override
+        public void handleCorrupted(ProcedureIterator procIter) throws IOException {
+          long corruptedCount = 0;
+          while (procIter.hasNext()) {
+            LOG.error("Corrupted procedure {}", procIter.next());
+            corruptedCount++;
+          }
+          if (corruptedCount > 0) {
+            throw new IOException("There are " + corruptedCount + " corrupted procedures when" +
+              " migrating from the old WAL based store to the new region based store, please" +
+              " fix them before upgrading again.");
+          }
         }
-        if (corruptedCount > 0) {
-          throw new IOException("There are " + corruptedCount + " corrupted procedures when" +
-            " migrating from the old WAL based store to the new region based store, please" +
-            " fix them before upgrading again.");
+      });
+      LOG.info("The WALProcedureStore max pid is {}, and the max pid of all loaded procedures is {}",
+        maxProcIdSet.longValue(), maxProcIdFromProcs.longValue());
+      // Theoretically, the maxProcIdSet should be greater than or equal to maxProcIdFromProcs, but
+      // anyway, let's do a check here.
+      if (maxProcIdSet.longValue() > maxProcIdFromProcs.longValue()) {
+        if (maxProcIdSet.longValue() > 0) {
+          // let's add a fake row to retain the max proc id
+          region.put(new Put(Bytes.toBytes(maxProcIdSet.longValue())).addColumn(FAMILY,
+            PROC_QUALIFIER, EMPTY_BYTE_ARRAY));
         }
+      } else if (maxProcIdSet.longValue() < maxProcIdFromProcs.longValue()) {
+        LOG.warn("The WALProcedureStore max pid is less than the max pid of all loaded procedures");
       }
-    });
-    LOG.info("The WALProcedureStore max pid is {}, and the max pid of all loaded procedures is {}",
-      maxProcIdSet.longValue(), maxProcIdFromProcs.longValue());
-    // Theoretically, the maxProcIdSet should be greater than or equal to maxProcIdFromProcs, but
-    // anyway, let's do a check here.
-    if (maxProcIdSet.longValue() > maxProcIdFromProcs.longValue()) {
-      if (maxProcIdSet.longValue() > 0) {
-        // let's add a fake row to retain the max proc id
-        region.put(new Put(Bytes.toBytes(maxProcIdSet.longValue())).addColumn(FAMILY,
-          PROC_QUALIFIER, EMPTY_BYTE_ARRAY));
+      if (!fs.delete(procWALDir, true)) {
+        throw new IOException("Failed to delete the WALProcedureStore migrated proc wal directory " +
+          procWALDir);
       }
-    } else if (maxProcIdSet.longValue() < maxProcIdFromProcs.longValue()) {
-      LOG.warn("The WALProcedureStore max pid is less than the max pid of all loaded procedures");
-    }
-    if (!fs.delete(procWALDir, true)) {
-      throw new IOException("Failed to delete the WALProcedureStore migrated proc wal directory " +
-        procWALDir);
+      LOG.info("Migration of WALProcedureStore finished");
+    } catch (IOException ioe) {
```

Comment:
Do we need the catch, or could just stop it on the finally and let the IOE be thrown upwards?

Reply:
Sorry @wchevreuil . I only now noticed you'd just put up a fix for the same thing.

On this line, I should have rethrown the caught exception. I wanted to pass in 'true' for abort if exception as opposed to 'false' for the finally block. I might be overthinking it since IIRC, if problem here, we'll crash out the Master.  Let me resolve this in favor of yours.